src/seals/base_envs.py

```diff
@@ -0,0 +1,160 @@
+"""Finite-horizon discrete environments with known transition dynamics."""
+
+import abc
+from typing import Optional
+
+import gym
+from gym import spaces
+import numpy as np
+
+from seals import util
+
+
+class ResettableEnv(gym.Env, abc.ABC):
```

Comment:
Hmm, I'm not sure what would be the best way of doing this - I guess you make a parameterized class or pass types in the constructor, but it is unclear for me whether pytype can use that - it would be better if we could infer types directly from the state and action spaces